### PR TITLE
fix(NODE-6348): Wrap thrown errors in JS Error objects with stacks

### PR DESCRIPTION
**Pull request by sophiebits**
We noticed in our usage of this package that thrown errors (eg: "Unknown frame descriptor") don't have a useful .stack property. This makes debugging harder. It seems this is expected behavior for errors created via napi at least on V8; it seems that the best way to remedy this is to recreate Error objects on the JS side?

### Double check the following

- [x] Ran `npm run format:js && npm run format:rs` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

**Comment by sophiebits** [comment]
I don't have a cargo setup on my machine at the moment but crossing my fingers the test passes in CI. :)

**Comment by addaleax** [comment]
Hi @sophiebits! 👋 CI's pointing out a syntax error in the test, let us know if you'd prefer to take care of that yourself or we should be pushing a fix this branch 🙂

**Comment by sophiebits** [comment]
Sloppy of me! Hope this will fix it. ^

File: test/index.test.js [hunk 1]
```diff
@@ -37,4 +37,17 @@ describe('zstd', () => {
...
     });
   });
+
+  describe('#decompress', () => {
+    context('when decompressing invalid data', () => {
+      it('includes a stack trace', async () => {
+        try {
+          await decompress(Buffer.from('invalid'));
+        } catch (error) {
+          expect(error.message).to.be('zstd: Unknown frame descriptor');
```
Comment (durran) [inline test/index.test.js:47]: We're using chai matchers here.

```suggestion
          expect(error.message).to.equal('zstd: Unknown frame descriptor');
```

File: test/index.test.js [hunk 2]
```diff
@@ -37,4 +37,17 @@ describe('zstd', () => {
...
+          expect(error.stack).to.match(/at decompress/);
```
Comment (durran) [inline test/index.test.js:48]: Stack will output with these changes:
```
Error: zstd: Unknown frame descriptor
    at module.exports.decompress (/Users/modetojoy/work/mongodb-js/zstd/index.js:167:11)
    at async Context.<anonymous> (/Users/modetojoy/work/mongodb-js/zstd/test/index.test.js:45:11)
```

```suggestion
          expect(error.stack).to.match(/module.exports.decompress/);
```

File: index.js [hunk 3]
```diff
@@ -151,5 +151,19 @@ if (!nativeBinding) {
 
 const { compress, decompress } = nativeBinding;
 
-module.exports.compress = compress;
-module.exports.decompress = decompress;
+// Error objects created via napi don't have JS stacks; wrap them so .stack is present
+// https://github.com/nodejs/node/issues/25318#issuecomment-451068073
+module.exports.compress = async function (data) {
```
Comment (durran) [inline index.js:156]: Note we will need to find another solution here. This file is generated by napi-rs so any `npm run build` is going to overwrite it.

Comment (sophiebits) [reply index.js:156]: hmm interesting. (sorry for missing that!)

do you have a suggestion for how best to fix? a second wrapper file I guess is one option but doesn't feel satisfying

**Review by sophiebits — COMMENTED** [review]
(sorry about the messy tests)

**Comment by sophiebits** [comment]
* updated to do the wrapping in a separate file
* updated so the names at runtime are like just `decompress` not `module.export.decompress`
* bindings have been regenerated; there are some minor changes since the last index.js update, perhaps because I'm on a newer napi? this is from napi 2.16.9. looks like mostly additional branches for possible architectures, not sure if this has a risk of breaking anything — lmk
* (got my setup fixed and verified the tests pass locally for me now!)

File: index.js (hunk 3 above)
Comment (durran) [reply index.js]: I think the approach you implemented is probably the best way to go for now. Thank you!

**Review by durran — APPROVED** [review]

**Review by durran — APPROVED** [review]
LGTM. Thanks for the contribution. I'll have the rest of the team look as well.

**Comment by durran** [comment]
Updated the bindings.js to the old index.js per discussion on internal Slack with respect to some operating systems we don't support (like Android). I've pushed that change to this PR.

**Review by addaleax — APPROVED** [review]

**Pull request merged**